EngineHacks/SkillSystem/Skills/StandaloneSkills/Discipline/Discipline.s

```diff
@@ -37,12 +38,21 @@ mov	lr, r2
 .short     0xF800
 orr	r0, r5
 cmp     r0, #0x0
-beq	NoSkill
+beq	NoDiscipline
 lsl     r4, r4, #0x1    @double WEXP if you have the skill
 cmp	r5, #0x0
+HasArmsMaster:
 bne 	DisciplinePlus
 
-NoSkill:
+NoDiscipline:
+@check for Arms Master
+ldr     r1, ArmsMasterID
+ldr     r2, SkillTester
+mov     r0, r7
+mov     lr, r2
+.short     0xF800    @two byte bl to lr
+cmp 	r0, #0x0
+bne 	HasArmsMaster
```

Comment:
You got a weird branch here that branches again as soon as the label is reached.

Reply:
Yeah, you're right. It would be the same thing if the jump after the check for the ArmsMaster skill goes straight to the DisciplinePlus label. I don't remember why I did it this way three weeks ago.